check.py

```diff
@@ -75,14 +82,44 @@ def fail_if_not_contained(actual, expected):
 else:
   tests = requested[:]
 
+warnings = []
+
 if not interpreter:
-  print 'warning: no interpreter provided (not testing spec interpreter validation)'
+  warnings.append('warning: no interpreter provided (did not test spec interpreter validation)')
 if not has_node:
-  print 'warning: no node found (not checking proper js form)'
+  warnings.append( 'warning: no node found (did not check proper js form)')
 if not has_mozjs:
-  print 'warning: no mozjs found (not checking asm.js validation)'
+  warnings.append( 'warning: no mozjs found (did not check asm.js validation)')
 if not has_emcc:
-  print 'warning: no emcc found (not checking emscripten/binaryen integration)'
+  warnings.append('warning: no emcc found (did not check non-vanilla emscripten/binaryen integration)')
+
+# setup
+
+BASE_DIR = os.path.abspath('test')
+WATERFALL_BUILD_DIR = os.path.join(BASE_DIR, 'llvm_install')
+
+def fetch_waterfall():
+  rev = open(os.path.join('test', 'revision')).read()
+  if os.path.exists(os.path.join(BASE_DIR, 'wasm-binaries-%s.tbz2' % rev)): return
+  # fetch it
+  print '(downloading waterfall ' + rev + ')'
+  basename = 'wasm-binaries-%s.tbz2'
+  downloaded = scripts.storage.download_tar(basename, BASE_DIR, rev)
+  scripts.support.untar(downloaded, WATERFALL_BUILD_DIR)
```

Comment:
`untar` expects to get the name that the tarfile untars into, I guess that's not sure intuitive: `wasm-install-$(build).tbz2` contains a single folder: `wasm-install`. The second parameter should be the path to where you want `wasm-install` to be, including `wasm-install`.

IIUC you want `WATERFALL_BUILD_DIR = os.path.join(BASE_DIR, 'wasm-install')` and then below `LLVM_DIR = os.path.abspath(os.path.join(WATERFALL_BUILD_DIR, 'bin'))` though I'd call it `BIN_DIR` instead since it contains more than LLVM.